Lib/tarfile.py

```diff
@@ -520,10 +516,7 @@ def read(self, size=None):
         """
         if size is None:
             t = []
-            while True:
-                buf = self._read(self.bufsize)
-                if not buf:
-                    break
+            while (buf := self._read(self.bufsize)):
```

Comment:
https://github.com/python/cpython/commit/8d130913cb9359c01de412178f9942419e921170  has eliminated this code (and just asserted that users will always call with a non-None size.)